metricbeat/module/apache/status/status.go

```diff
@@ -31,51 +30,49 @@ var (
 )
 
 func init() {
-	if err := mb.Registry.AddMetricSet("apache", "status", New); err != nil {
+	if err := mb.Registry.AddMetricSet("apache", "status", New, ParseServerStatusURL); err != nil {
 		panic(err)
 	}
 }
 
-// MetricSet for fetching Apache HTTPD server status.
-type MetricSet struct {
-	mb.BaseMetricSet
-	client *http.Client // HTTP client that is reused across requests.
-	url    string       // Apache HTTP server status endpoint URL.
-}
-
-// New creates new instance of MetricSet.
-func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
-	// Additional configuration options
+// ParseServerStatusURL takes the configuration data and host value and
+// constructs a HostInfo object containing the URI to the Apache server status
+// endpoint.
+func ParseServerStatusURL(mod mb.Module, host string) (mb.HostInfo, error) {
 	config := struct {
```

Comment:
I kind of liked that config part was in `New`. But I see that having it here will make it easier reusable for the different modules.